test/github-issues/10121/entity/estate.ts

```diff
@@ -0,0 +1,40 @@
+import {
+    Column,
+    Entity,
+    JoinColumn,
+    OneToOne,
+    PrimaryGeneratedColumn,
+} from "../../../../src"
+import CityModel from "./city"
+import PrefModel from "./pref"
+
+@Entity({ name: "M_ESTATE" })
+export default class EstateModel {
+    @PrimaryGeneratedColumn({ name: "ESTATE_ID" })
+    id!: number
+
+    @Column({
+        name: "ESTATE_NAME",
+        type: "varchar",
+        length: 100,
+        nullable: false,
+    })
+    name!: string
+
+    @Column({ name: "PREF_ID", comment: "都道府県ID" })
+    prefId!: string
+
+    @Column({ name: "CITY_ID", type: "varchar", comment: "市区ID" })
+    cityId!: string | null
```

Comment:
`!` and `| null` at the same time? They contradict each other.

I would not add `!` anywhere in the entities. If the column is nullable, I would defined it like `cityId: string | null`. If it is not nullable, then `cityId: string` (especially in this project since `strictNullChecks` are enabled.

Reply:
In my projects I do exactly as you write.
This code is copied from issue #10121